x-pack/plugins/lens/public/drag_drop/drag_drop.tsx

```diff
@@ -247,4 +267,188 @@ const DragDropInner = React.memo(function DragDropInner(
+    itemsInGroup?.length &&
+    itemsInGroup.length > 1 &&
+    value?.id
+  ) {
+    element = (
+      <ReorderableDragDrop
+        draggingProps={{
+          className: classNames(children.props.className, classes),
+          draggable,
+          onDragEnd: dragEnd,
+          onDragStart: dragStart,
+          ['data-test-subj']: props['data-test-subj'] || 'lnsDragDrop',
+        }}
+        dropProps={{
+          onDrop: drop,
+          onDragOver: dragOver,
+          onDragLeave: dragLeave,
+          dragging,
+          droppable,
+          itemsInGroup,
+          id: value.id,
+        }}
+      >
+        {children}
+      </ReorderableDragDrop>
+    );
+  }
+
+  if (!itemsInGroup?.length || itemsInGroup.length < 2 || !value || !dropTo || !draggable) {
+    return element;
+  }
+  const groupLength = itemsInGroup.length;
+  const currentIndex = itemsInGroup.indexOf(value.id);
+
+  const { label } = props as DraggableProps;
+
+  return (
+    <>
+      <EuiScreenReaderOnly>
+        <button
+          aria-label={
+            !state.keyboardMode
+              ? i18n.translate('xpack.lens.dragDrop.reorderInactive', {
+                  defaultMessage:
+                    '{label} is in position {position} of {groupLength}. Press enter/space to initiate re-order. Then use the up/down arrow keys to re-order.',
+                  values: {
+                    label,
+                    position: currentIndex + 1,
+                    groupLength,
+                  },
+                })
+              : i18n.translate('xpack.lens.dragDrop.reorderActive', {
+                  defaultMessage:
+                    '{label} is in position {position} of {groupLength}. Use the up/down arrow keys to re-order. Press escape to leave re-ordering.',
+                  values: {
+                    label,
+                    position: currentIndex + 1,
+                    groupLength,
+                  },
+                })
+          }
+          onKeyDown={(e: React.KeyboardEvent<HTMLButtonElement>) => {
+            if (e.key === keys.ENTER || e.key === keys.SPACE) {
+              setState({ ...state, keyboardMode: !state.keyboardMode });
+            } else if (e.key === keys.ESCAPE) {
+              setState({ ...state, keyboardMode: false });
+            }
+            if (state.keyboardMode) {
+              e.stopPropagation();
+              e.preventDefault();
+
+              if (keys.ARROW_DOWN === e.key) {
+                if (currentIndex < groupLength - 1) {
+                  dropTo(itemsInGroup[currentIndex + 1]);
+                }
+              } else if (keys.ARROW_UP === e.key) {
+                if (currentIndex > 0) {
+                  dropTo(itemsInGroup[currentIndex - 1]);
+                }
+              }
+            }
+          }}
+        >
+          {i18n.translate('xpack.lens.dragDrop.screenReaderButtonLabel', {
+            defaultMessage: 'Press enter to reorder (copy pending)',
```

Comment:
Is "(copy pending)" supposed to be here?

Reply:
No, I haven't received enough feedback about accessibility and we're still in the process of forming the right copy. I'll come up with a proposal myself today and we'll see how it goes.